lib/util.js

```diff
@@ -345,11 +349,31 @@ function getSystemErrorName(err) {
   return internalErrorName(err);
 }
 
+function addSafeAbortSignalAbortListener(signal, listener, options = kEmptyObject) {
+  if (signal === undefined) {
+    throw new ERR_INVALID_ARG_TYPE('signal', 'AbortSignal', signal)
+  }
+  validateAbortSignal(signal, 'signal');
+  kResistStopPropagation ??= require('internal/event_target').kResistStopPropagation;
+  signal.addEventListener('abort', listener, { __proto__: null, ...options, [kResistStopPropagation]: true });
+  const removeEventListener = () => {
+    signal.removeEventListener('abort', listener, options);
+  };
+  return {
+    __proto__: null,
+    removeEventListener,
```

Comment:
I am debating if we should export this along with `SymbolDispose` as the `using` keyword is not yet available 🤔

Reply:
why is `SymbolDispose` relevant here?

Reply:
It allows explicitly removing the event listener once there is no need for it anymore (which means clearing the pointer to `signal` added by the event listener)

Reply:
https://github.com/tc39/proposal-explicit-resource-management/blob/3b576c0bb756fcbb4b91c5556c3609806f9abaf9/README.md?plain=1#L1731-L1733